libs/irSensor/sim/state.ts

```diff
@@ -0,0 +1,38 @@
+/// <reference path="../../../built/common-sim.d.ts"/>
+
+namespace pxsim {
+    export interface IrSensorBoard extends CommonBoard {
+        irSensorState: IrSensorState;
+    }
+
+    export function irSensorState(): IrSensorState {
+        return (board() as IrSensorBoard).irSensorState;
+    }
+
+    export class IrSensorState {
+        public sensorUsed: boolean = false;
```

Comment:
You need to  by use a level detector to generate the hot/cold events

Reply:
This class is just test for **Ir Sensor** simulation.
I'm not sure whether to generate the hot/cold events.